src/test/crossFetchNode.test.ts

```diff
@@ -273,3 +273,131 @@ test('should not fail when arbitrary parameters are configured in fetchOptions',
+    .reply(
+      200,
+      {content: 'not empty'},
+      {'content-type': 'application-json charset=UTF-8'}
+    );
+
+  const client = new ShopperSearch({...config});
+  const response = await client.productSearch({
+    parameters: {q: 'sony'},
+    headers: {authorization: 'Bearer test-auth'},
+  });
+
+  expect(response).toEqual({content: 'not empty'});
+});
+
+test('do not throw in 2xx response', async () => {
+  nock('https://localhost:3000')
+    .get(
+      `/search/shopper-search/v1/organizations/${config.parameters.organizationId}/product-search?siteId=${config.parameters.siteId}&q=sony`
+    )
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(
+      212,
+      {content: 'not empty'},
+      {'content-type': 'application-json charset=UTF-8'}
+    );
+
+  const client = new ShopperSearch({...config});
+  const response = await client.productSearch({
+    parameters: {q: 'sony'},
+    headers: {authorization: 'Bearer test-auth'},
+  });
+
+  expect(response).toEqual({content: 'not empty'});
+});
+
+test('do not throw in 304 response', async () => {
+  nock('https://localhost:3000')
+    .get(
+      `/search/shopper-search/v1/organizations/${config.parameters.organizationId}/product-search?siteId=${config.parameters.siteId}&q=sony`
+    )
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(
+      304,
+      {content: 'not empty'},
+      {'content-type': 'application-json charset=UTF-8'}
+    );
+
+  const client = new ShopperSearch({...config});
+  const response = await client.productSearch({
+    parameters: {q: 'sony'},
+    headers: {authorization: 'Bearer test-auth'},
+  });
+
+  expect(response).toEqual({content: 'not empty'});
+});
+
+test('do not throw in empty body', async () => {
+  nock('https://localhost:3000')
+    .get(
+      `/search/shopper-search/v1/organizations/${config.parameters.organizationId}/product-search?siteId=${config.parameters.siteId}&q=sony`
+    )
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(200);
+
+  const client = new ShopperSearch({...config});
+  const response = await client.productSearch({
+    parameters: {q: 'sony'},
+    headers: {authorization: 'Bearer test-auth'},
+  });
+
+  expect(response).toEqual({});
+});
+
+test('handle void methods', async () => {
+  nock('https://localhost:3000')
+    .post(
+      `/customer/shopper-customers/v1/organizations/${config.parameters.organizationId}/customers/password/actions/reset`
+    )
+    .query({
+      siteId: config.parameters.siteId,
+    })
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(204);
+
+  const client = new ShopperCustomers({...config});
+  const response = await client.resetPassword({
+    headers: {
+      authorization: 'Bearer test-auth',
+    },
+    body: {
+      resetToken: 'R1e2s3e4t5T6o7k8e9n0',
+      login: 'janedoe@test.com',
+      newPassword: 'p@assword2',
+    },
+  });
+  expect(response);
```

Comment:
```suggestion
  expect(response).toBeUndefined();
```
(Might not be the actual assertion, but I think it is...)

Reply:
👌

Reply:
It is the assertion :)